llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -7651,6 +7651,12 @@ bool AMDGPULegalizerInfo::legalizeIntrinsic(LegalizerHelper &Helper,
     return legalizeLaneOp(Helper, MI, IntrID);
   case Intrinsic::amdgcn_s_buffer_prefetch_data:
     return legalizeSBufferPrefetch(Helper, MI);
+  case Intrinsic::amdgcn_dead: {
+    for (const MachineOperand &Def : MI.defs())
```

Comment:
How can this intrinsic get multiple defs in the first place?

Reply:
It can't

Reply:
It can when we return a struct type.